src/index.ts

```diff
@@ -20,24 +23,49 @@ const ID_MULTIHASH_CODE = identity.code
 export const namespace = '/ipns/'
 export const namespaceLength = namespace.length
 
-export interface IPNSEntry {
-  value: Uint8Array
-  signature: Uint8Array // signature of the record
-  validityType: IpnsEntry.ValidityType // Type of validation being used
-  validity: Uint8Array // expiration datetime for the record in RFC3339 format
-  sequence: bigint // number representing the version of the record
-  ttl?: bigint // ttl in nanoseconds
-  pubKey?: Uint8Array // the public portion of the key that signed this record (only present if it was not embedded in the IPNS key)
-  signatureV2?: Uint8Array // the v2 signature of the record
-  data?: Uint8Array // extensible data
-}
+export class IPNSRecord {
+  readonly pb: IpnsEntry
```

Comment:
Why do we need to expose the serialization format?  This module exists to deal with serialization/deserialization - it seems a bit strange to expose the intermediate format to the world.

Reply:
This was mostly for testing purposes. But that'll be changed when refactoring to the interface - in 2 weeks.